onnxruntime/core/providers/tensorrt/onnx_ctx_model_helper.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include <filesystem>
+
+#include "NvInfer.h"
+#include "core/providers/shared_library/provider_api.h"
+
+namespace onnxruntime {
+
+static const std::string EPCONTEXT_OP = "EPContext";
+static const std::string MAIN_CONTEXT = "main_context";
+static const std::string EMBED_MODE = "embed_mode";
+static const std::string EP_CACHE_CONTEXT = "ep_cache_context";
```

Comment:
Considering the types of AttributeProto that ONNX supports, I'm afraid that the string is the most suitable one for the ep_cache_context.
For those types that won't be hashed in ORT, we can't use AttributeProto.GRAPH since it's not compatible with TRT engine cache. I thought about AttributeProto.UNDEFINED, but [make_attribute](https://github.com/onnx/onnx/blob/main/onnx/helper.py#L838) in ONNX doesn't handle the undefined type case. So, the script can't generate the attribute with undefined type.
Therefore, i think setting a low optimization level in session option is the better option for now.

Reply:
Ok understood. We should add this to the docs that it is preferred to store a path to an engine rather than storing the actual engine inside the ONNX.